### DISCUSSION TITLE
Sorting formatted numbers in a table

### DISCUSSION THREAD

**Discussion by cypherpork** [sqlpage/SQLPage#564] [Q&A]
The following sequel:
```sql
SELECT
	'table' AS component,
	'Name' AS markdown,
	"dailyusers" AS id,
	TRUE AS sort,
	TRUE AS small;
SELECT daily_users AS "Users" 
FROM   tb_users;
```
Produces a sortable table:
| USERS  |
| ------ |
| 110838 |
| 107439 |
| 97411  |
| 90563  |
| 14840  |
| 8355   |
| 6527   |
| 5617   |
| 5366   |

I want to format the numbers with thousands separators:
```sql
SELECT
	'table' AS component,
	'Name' AS markdown,
	"dailyusers" AS id,
	TRUE AS sort,
	TRUE AS small;
SELECT format( daily_users, 0 ) AS "Users" 
FROM   tb_users;
```
which produces the following for the US locale:
| USERS  |
| ------ |
| 110,838 |
| 107,439 |
| 97,411  |
| 90,563  |
| 14,840  |
| 8,355   |
| 6,527   |
| 5,617   |
| 5,366   |

Now, the numbers are treated as text and slpage sorts them incorrectly.

Does anyone know an easy way to fix it?

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage !

Unfortunately, there is no builtin solution to this, but you can use javascript to format the numbers after the page has and the table has been indexed.

```js
document.querySelectorAll('td.users').forEach(x => x.textContent = parseFloat(x.textContent).toLocaleString()); 
```

You can use the javascript property of the shell component to include this custom script.
